ts/features/pn/components/PnServiceCTA.tsx

```diff
@@ -110,12 +110,11 @@ const PnServiceCTA = ({ serviceId, activate }: Props) => {
     setIsUpdating(isStillUpdating);
   }, [isUpdating, dispatch, serviceId, serviceActivation]);
 
-  useOnFirstRender(
-    () => {
+  useOnFirstRender(() => {
```

Comment:
In this case `typeof activate` is giving me a `string`. This means that there is a wrong type somewhere up in the chain. To fix this behaviour completely it's possible to add:

```ts
const activate = (props.route.params.activate as unknown) === "true";
```

inside this file: https://github.com/pagopa/io-app/blob/IAMVL-95-fix-auto-upsert/ts/screens/services/ServiceDetailsScreen.tsx#L236 (in the future we could replace it with https://gcanti.github.io/io-ts-types/modules/BooleanFromString.ts.html from `io-ts`).

With this fix you should be able to use this hook correctly like:

```ts
useOnFirstRender(
    () => {
      dispatch(pnActivationUpsert.request(true));
    },
    () => activate === true
  );
```

What do you think? 🙂

Reply:
@dgopsq what would happen if I pass it as a normal navigation param (instead of passing through an internal link)? In that case it should have the correct boolean type, shouldn't it?